services/motion/builtin/state/state.go

```diff
@@ -52,16 +67,523 @@ type PlannerExecutor interface {
 	Cancel()
 }
 
+type componentState struct {
+	executionIDHistory []motion.ExecutionID
+	executionsByID     map[motion.ExecutionID]stateExecution
+}
+
+type newPlanMsg struct {
+	plan       motion.Plan
+	planStatus motion.PlanStatus
+}
+
+type stateUpdateMsg struct {
+	componentName resource.Name
+	executionID   motion.ExecutionID
+	planID        motion.PlanID
+	planStatus    motion.PlanStatus
+}
+
+// a stateExecution is the struct held in the state that
+// holds the history of plans & plan status updates an
+// execution has exprienced & the waitGroup & cancelFunc
+// required to shut down an execution's goroutine.
+type stateExecution struct {
```

Comment:
can we call this `execution` instead?  What does the word state distinguish here??

Reply:
This is the struct that lives in the State struct. This is opposed to the `execution[R any]` which only ever exists in the StartExecution function & the go routine it creates.